docs/features/pixi_global_manifest.md

```diff
@@ -0,0 +1,143 @@
+# Pixi Global Manifest
+
+### Motivation
+
+`pixi global` is currently limited to imperatively managing CLI packages.
+The next iteration of this feature should fulfill the following needs:
+
+- Shareable global environments.
+- Allow managing complex environments with multiple packages as dependencies
+- Flexible exposure of binaries
+
+## Design Considerations
+
+There are a few things we wanted to keep in mind in the design:
+
+1. **User-friendliness**: Pixi is a user focussed tool that goes beyond developers. The feature should have good error reporting and helpful documentation from the start.
+2. **Keep it simple**: The CLI should be all you need to interact with the global environments.
+3. **Unsurprising**: Simple commands should still behave similar to traditional package managers.
+4. **Human Readable**: Any file created by this feature should be human-readable and modifiable.
+
+## Manifest
+
+```toml title="pixi-global.toml"
+# `python_3_10` is the name of the environment
+[envs.python_3_10.dependencies.python]
+spec = "3.10.*"
+expose_binaries = {"python3.10"="python"} # `python` from `python_3_10` will be available as `python3.10`
+
+[envs.python.dependencies.python]
+spec = "3.11.*"
+expose_binaries = "auto" # This will expose python, python3 and python3.11
+
+[envs.python.dependencies.pip]
+spec = "*"
+```
+
+## CLI
+
+Install one or more packages `PACKAGE` into their own environments and expose their binaries
+
+```
+pixi global install [PACKAGE]...
+```
+
+Remove environments of the named packages `PACKAGE`
+```
+pixi global uninstall [PACKAGE]...
+```
+
+Resolve environment without previously defined specs
+```
+pixi global upgrade python
+```
+
+Add package `PACKAGE` to environment `ENV`.
+If environment `ENV` does not yet exist, then it will be created.
+```
+pixi global add --name ENV PACKAGE
+```
+
+Remove package `PACKAGE` from environment `ENV`.
+```
+pixi global remove --name ENV PACKAGE
+```
+
+Update the version of one package
+```
+pixi global update --name python python=3.12.*
+```
+
+Set for a specific package `PACKAGE` in environment `ENV` under which `MAPPING` binaries are exposed
+```
+pixi expose-bin --name ENV --package PACKAGE [MAPPING]...
+```
+
+Ensure that the environments on the machine reflect the state in the manifest
```

Comment:
I think we also have a situation where a manifest does not exist, but some of the envs are already created. 
What would be the behaviour `pixi global sync`? 
should we for first iteration throw and error but and later add the reverse sync 
machine --> manifest?

Reply:
I think every pixi global command should offer to create a new manifest. If there are environments present, pixi should offer to create a manifest that matches those environments.

Reply:
There is not a perfect way to do it from the current environments, but using the information in the `.pixi/bin` and the `.pixi/envs/xxx/conda-meta` you can easily derive the manifest information. 

So I would be for having `pixi global sync` work, by making sure the environments are setup as defined in the manifest. Except when there is no manifest, it allows it to be generated from the current state. 

Most `pixi global` sub commands imply a `pixi global sync`.

- Users should be able to change the manifest by hand (creating or modifying (adding or removing))
- Users should be able to "export" their existing environments into the manifest, if non existing.
- The manifest is always "in sync" after an `install/remove/inject/other` global command.

```
# first time, clean computer:
> pixi global install python
# creates manifest and .pixi/envs/python
> rm `~/.pixi`
> pixi global sync
# install python in `~/.pixi/envs/python`

> rm <manifest>
> pixi global sync
# Creates manifest and add environment python to the manifest
# Interactively or not TBD.

# removed the python environment from the manifest
> vim <manifest> 
> pixi global sync
# Removes the `~/.pixi/envs/python` environment from the file system.
```